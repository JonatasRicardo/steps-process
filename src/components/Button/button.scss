@import '../../theme.scss';

.button {
    cursor: pointer;
    border: 0;
    border-radius: 4px;
    margin-left: 5px;
    background-color: $colorBlue;
    color: #ffffff;
    padding: 10px 18px;
    font-size: 15px;
    font-weight: 100;
    transition: all .5s;
    box-shadow: 0px 0px 15x -13px rgba(0,0,0,0.75);
    transition: all .5s;

    &:hover{
        box-shadow: 0px 0px 15px -2px $colorBlue;
    }
    &:active{
        box-shadow: inset 0px 0px 15px -2px rgba(0,0,0,0.3);

    }

    &.light {
        background-color: $colorWhite;
        color: $colorBlue;
        border: 1px solid $colorGrayLight;
    }
}