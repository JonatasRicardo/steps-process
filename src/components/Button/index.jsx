import React from 'react';
import PropTypes from 'prop-types';
import './button.scss';

const propTypes = {
    children: PropTypes.oneOfType([
        PropTypes.arrayOf(PropTypes.node),
        PropTypes.node
    ]),
    light: PropTypes.bool,
}

const defaultProps = {
    light: false,
}

function Button(props) {
    const { children, light } = props;
    const lightCLass = light ? 'light' : '';
    return (
        <button className={`button ${lightCLass}`} data-testid="button" {...props}>
            {children}
        </button>
    );
}

Button.propTypes = propTypes;
Button.defaultProps = defaultProps;

export default Button;