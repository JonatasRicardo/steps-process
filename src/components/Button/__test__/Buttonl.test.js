import React from 'react';
import { render } from '@testing-library/react';
import Button from '../index';

describe('<Button/>', () => {
  it('renders Button components', () => {
    const { getByTestId } = render(
      <Button />
    );
    const buttonElement = getByTestId('button');
    expect(buttonElement).toBeDefined();
  });
  it('Button children', () => {
    const { getByText } = render(
      <Button>
        Click me
      </Button>
    );
    const buttonElement = getByText('Click me');
    expect(buttonElement).toBeDefined();
    expect(buttonElement).toMatchSnapshot();
  });
  it('Button light', () => {
    const { getByTestId } = render(
      <Button light>
        Click me
      </Button>
    );
    const buttonElement = getByTestId('button');
    expect(buttonElement).toHaveClass('light');
    expect(buttonElement).toMatchSnapshot();
  });


})
