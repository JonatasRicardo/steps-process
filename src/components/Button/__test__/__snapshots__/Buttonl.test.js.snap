// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button/> Button children 1`] = `
<button
  class="button "
  data-testid="button"
>
  Click me
</button>
`;

exports[`<Button/> Button light 1`] = `
<button
  class="button light"
  data-testid="button"
>
  Click me
</button>
`;
